wallet/src/de/schildbach/wallet/ui/main/TransactionAdapter.kt

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.main
+
+import android.content.res.Resources
+import android.text.format.DateUtils
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.ui.transactions.TransactionRowView
+import de.schildbach.wallet.ui.transactions.TxResourceMapper
+import de.schildbach.wallet_test.R
+import de.schildbach.wallet_test.databinding.TransactionRowBinding
+import org.bitcoinj.core.*
+import org.bitcoinj.utils.ExchangeRate
+import org.bitcoinj.utils.MonetaryFormat
+import org.dash.wallet.common.ui.getRoundedBackground
+import org.dash.wallet.common.ui.getRoundedRippleBackground
+import org.dash.wallet.common.util.GenericUtils
+
+class TransactionAdapter(
+    private val dashFormat: MonetaryFormat,
+    private val resources: Resources,
+    private val drawBackground: Boolean = false,
+    private val clickListener: (TransactionRowView, Int) -> Unit
+) : ListAdapter<TransactionRowView, TransactionAdapter.TransactionViewHolder>(DiffCallback()) {
+    private val contentColor = resources.getColor(R.color.content_primary, null)
+    private val warningColor = resources.getColor(R.color.content_warning, null)
+    private val colorSecondaryStatus = resources.getColor(R.color.secondary_status, null)
+
+    class DiffCallback : DiffUtil.ItemCallback<TransactionRowView>() {
```

Comment:
In DashPay there is another TransactionAdapter class that is used to show transactions that are between two users.  That transaction list contains other items besides transactions too.

Reply:
Right, in DashPay the old adapter is extended to represent other types of rows in the list.
Since we have a view representation here now (`TransactionRowView`), it can be easily extended to represent other types of rows and the adapter can switch between view holders depending on the type. I think merging this will be some work, but easier than might seem.